src/App.js

```diff
@@ -61,16 +110,43 @@ function App() {
   }
 
   //Basic CRUD-Operations, used on CreateGuest.js and EditGuest.js
-  function createGuest(newName, intolerancesArray, newNotes) {
-    setGuestArray([
+  async function createGuest(newName, intolerancesArray, newNotes) {
+    const newGuestArray = [
       {
         name: newName,
         intolerances: intolerancesArray,
         notes: newNotes,
         selected: false,
       },
       ...guestArray,
-    ]);
+    ];
+    //it's necessary to save the new guest to the db, so that an _id is generated directly:
+    const response = await fetch(`/api/users/addGuest/${username}`, {
+      method: "PUT",
+      body: JSON.stringify({ guestList: newGuestArray }),
+      headers: {
+        "Content-Type": "application/json",
+      },
+    });
+    const json = await response.json();
+
+    if (!response.ok) {
+      console.error(json.error);
+    }
+    if (response.ok) {
+      setIsChanges(false);
+    }
+    //fetch the newly created guest from db and save him in guestArray:
+    try {
+      const response = await fetch(`/api/users/${username}`);
+      const data = await response.json();
+
+      if (response.ok) {
+        setGuestArray([...data[0].guestList]);
+      }
+    } catch (error) {
+      console.error(error.message);
+    }
```

Comment:
Why do you need to fetch the newly created guest here? Can't you just set the local state?

Reply:
That's because every guest gets the _id-property from the db. I use the _id then to toggle, edit, select... So if I don't fetch the newly created guest here, all of those functions break